### PR TITLE
[9.0][MIG] medical patient dob

### PR DESCRIPTION
**Pull request by abigailfoster**
This is dependent on #69 

This module appends the patient's date of birth to their name when displayed in a search field autocomplete.
- [X] #69

### CODE REVIEW

**Comment by lasley** [comment]
@jamesfoster85 and I reviewed this one together in a side conversation. I give a :+1:

**Comment by pedrobaeza** [comment]
I will review it when the dependency is merged and this PR rebased so that it's easier to do it. Let's focus in reviewing the base modules first.

**Comment by lasley** [comment]
No prob, we're doing our side-reviews just so we can meet the deadlines & just dropping the modules as PRs here for finalization after we merge into our internal release branch. I'll be applying your comments from the base module onto the others ones later today. Base should be good.

**Comment by woodbrettm** [comment]
Ready for review.

File: medical_patient_dob/README.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+This module adds date of birth to patient name in search field autocomplete.
+
+Usage
+=====
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/159/9.0
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
```
Comment (lasley) [inline medical_patient_dob/README.rst]: Remove these lines

File: medical_patient_dob/models/medical_patient.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+class MedicalPatient(models.Model):
+    _inherit = 'medical.patient'
+
+    @api.multi
+    def _format_dob(self):
+        self.ensure_one()
+        date = 'No DoB'
+        if self.dob:
+            date = fields.Datetime.from_string(self.dob).strftime('%m/%d/%Y')
```
Comment (lasley) [inline medical_patient_dob/models/medical_patient.py]: Hmmmm this isn't internationalized - Europe reverses the month and day aspects of this.

I feel like we should maybe use this instead: [convert_to_display_name](https://github.com/odoo/odoo/blob/9.0/openerp/fields.py#L1438). Pretty sure that will use the internal internationalization settings in order to display proper.

**Review by lasley — CHANGES_REQUESTED** [review]

**Comment by woodbrettm** [comment]
Updated. 
- Tests also pass when language in the UI is set to English UK (dd/mm/yyyy) for example and when language is set to None. 
- Another test was added to account for changes in date format

**Review by lasley — APPROVED** [review]
Thanks @bmw95 

@t3ddftw please review

**Review by tedsalmon — APPROVED** [review]
👍 - Looks good!

**Pull request merged**